client/src/components/product_overview/Overview.jsx

```diff
@@ -30,16 +30,18 @@ class Overview extends React.Component {
         this.setState({ images: imagesArray });
```

Comment:
you can call setState once here and handle all the changes you doing in one setState method